src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -0,0 +1,473 @@
+(ns quo2.components.record-audio.record-audio.view
+  (:require [quo.react :refer [memo effect!]]
+            [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.icon :as icons]
+            [quo2.components.buttons.button :as button]
+            [react-native.reanimated :as reanimated]
+            [reagent.core :as reagent]
+            [cljs-bean.core :as bean]
+            [quo2.components.record-audio.record-audio.style :as style]
+            [status-im.utils.utils :as utils]))
+
+(defonce recording? (reagent/atom false))
```

Comment:
This is a lot of global state @briansztamfater, it makes reasoning about the code even harder because any function can access anything. Have you considered using local component state?

IMO we should almost never use this kind of approach to manage state. Even in pure React land we don't see this, because hooks are always used inside components and follow the normal rendering lifecycle.

Notice that by using global state we're limiting pretty much all components in this namespace to be only used once per screen. It also makes testing much more fragile (in case we want to write component tests in the future), because every test will need to reset lots of global state, otherwise we get flaky tests.

On top of that it makes using the REPL less enjoyable in my experience, because now you need to resort to different tricks to reset the state, redefine `defonce`, etc.

Reply:
Thanks for pointing this out. I've tried both ways through development iteration, didn't have clear all the pros & cons of both approaches. I moved ratoms / atoms to the component itself.